pyccel/codegen/printing/ccode.py

```diff
@@ -972,6 +974,16 @@ def _print_Assign(self, expr):
             return '{};'.format(self._print(expr.rhs))
         lhs = self._print(expr.lhs)
         rhs = expr.rhs
+        if isinstance(rhs, IndexedElement) and isinstance(expr.lhs, Variable):
+            free_code = ''
+            for i in rhs.args:
+                if isinstance(i, Slice):
+                    free_code += 'free_array(%s);\n' % self._print(expr.lhs.name)
+                    self._additional_imports.add('ndarrays')
+                    break
```

Comment:
This can be written more succinctly as:
```
if any(isinstance(i, Slice) for i in rhs.args):
    free_code += 'free_array(%s);\n' % self._print(expr.lhs.name)
    self._additional_imports.add('ndarrays')
```
One line loops are always much faster than multi-line loops in python and `break` statements tend to make code harder to understand